mongodb/src/test/scala/akka/stream/alpakka/mongodb/MongoSourceSpec.scala

```diff
@@ -25,6 +28,11 @@ class MongoSourceSpec
     with BeforeAndAfterAll
     with MustMatchers {
 
+  // case class and codec for mongodb macros
+  case class Number(_id: Int)
+
+  val codecRegistry = fromRegistries(fromProviders(classOf[Number]), DEFAULT_CODEC_REGISTRY)
```

Comment:
Shouldn't this also be shown in the sample?

Reply:
can you point me to this sample ?

Reply:
Ah, right. I think we should leave MongoSourceSpec as it was (using Document in all test cases) and create a new MongoSourceWithCodecSpec using the Number case class in all test cases.
This way we can easily refer to this new optional feature in the docs and leave the existing basic one as is. Sounds good @vgkowski ?

Reply:
This is the example @jochenschneider is talking about: https://developer.lightbend.com/docs/alpakka/latest/mongodb.html#usage

The docs will show the code inside the #init-connection tags in MongoSourceSpec. Since we've added ".withCodecRegistry(codecRegistry)", readers will be left wondering where is the _codecRegistry_ coming from.

Reply:
Let's try this ? https://github.com/vgkowski/alpakka/tree/pr796